### ISSUE TITLE
Context variables are not available in functions written as separate files when using Stepwise Planner 

### ISSUE THREAD

**Issue by jasonjohnlee** [microsoft/semantic-kernel#3710]
**Describe the bug**
I have a similar bug to #2906 but with [functions defined in files](https://learn.microsoft.com/en-us/semantic-kernel/ai-orchestration/plugins/semantic-functions/serializing-semantic-functions?tabs=Csharp).  Context variables work fine with the semantic function when I use a Sequential Planner.   When I switch to use a Stepwise Planner, the context variables appear to be null even though I'm using almost exactly the same code.

**To Reproduce**
1. Create a plugin with a function defined in a file.  Use a context variable in that function as circled below.
![image](https://github.com/microsoft/semantic-kernel/assets/16003226/9c8d2b53-f6b4-4821-8307-fa407d4509af)
![image](https://github.com/microsoft/semantic-kernel/assets/16003226/e7f4e868-cf1f-4dff-af70-2139d10c32c6)
1. Load that plugin into the Kernel
![image](https://github.com/microsoft/semantic-kernel/assets/16003226/7a3be35b-1f0f-4850-aa49-63a6aa61074d)
1. Create a context with variables, create a StepwisePlanner plan, and invoke.  Note that the commented out SequentialPlanner code results in the expected behaviour. 
![image](https://github.com/microsoft/semantic-kernel/assets/16003226/94dcac02-5e76-4748-a4f8-b187af486a4c)
1. The final prompt sent to Azure Open AI does not contain the "questionContext" variable value. 
![image](https://github.com/microsoft/semantic-kernel/assets/16003226/8f58d315-7a27-4f39-b861-e42aa9fd944a)

**Expected behavior**
I expect the prompt sent to Azure Open AI to look like how it looks when using StepwisePlanner.  Here is a screenshot using the commented out code in an earlier screenshot.
![image](https://github.com/microsoft/semantic-kernel/assets/16003226/397d6401-8a48-4475-a4a9-f3dccecb1b3a)

**Platform**
 - OS: Windows
 - IDE: Visual Studio 2022
 - Language: C#
 - Source: 1.0.0-beta8

**Comment by matthewbolanos** [comment]
With something like the stepwise planner, the planner should be the one that sets the input variables for functions (e.g., `questionContext`). Can you confirm that `questionContext` is being defined as an `input_variable` for your semantic function?

**Comment by jasonjohnlee** [comment]
> With something like the stepwise planner, the planner should be the one that sets the input variables for functions (e.g., `questionContext`). Can you confirm that `questionContext` is being defined as an `input_variable` for your semantic function?

Thanks for the suggestion @matthewbolanos, but unfortunately I still get this behaviour with _questionContext_ defined in my input parameters.  I've tried it both with and without _questionContext_ in my _config.json_.  Here's the input parameters section of my _config.json_ at the moment.

```
  "input": {
    "parameters": [
      {
        "name": "input",
        "description": "The question the user is asking.",
        "defaultValue": ""
      },

      {
        "name": "questionContext",
        "description": "Private information specific to trip that could help answer the question.",
        "defaultValue": ""
      }
    ]
```

**Comment by matthewbolanos** [comment]
Would you be able to confirm if you're still having this issue with the Function Calling Stepwise Planner?

**Comment by jasonjohnlee** [comment]
@matthewbolanos That is correct, I haven't resolved this problem yet.  At the end of December, I tried upgrading to version 1.0.1 and I noticed that StepwisePlanner has been deprecated and replaced by an _IChatCompletionService_ with _AutoInvokeKernelFunctions_.  I gave that a try, but I have a different problem where it only seems to execute one function no matter how I changed the descriptions of the functions I wanted the planner to execute.  I decided to shelve my project until more documentation and guidance on planners is available since I may be implementing this functionality incorrectly.  There isn't much documentation on how to do this properly when I last looked.

**Comment by markwallace-microsoft** [comment]
All .Net issues prior to 1-Dec-2023 are being closed. Please re-open, if this issue is still relevant to the .Net Semantic Kernel 1.x release. In the future all issues that are inactive for more than 90 days will be labelled as 'stale' and closed 14 days later.
